internal/templates/01-update.yaml.tmpl

```diff
@@ -1,14 +1,56 @@
 # This file belongs to the resource update step.
-apiVersion: kuttl.dev/v1beta1
-kind: TestStep
-commands:
-{{- range $resource := .Resources }}
-{{- if eq $resource.KindGroup "secret." -}}
-  {{continue}}
-{{- end -}}
-{{- if not $resource.Namespace }}
-{{- if $resource.Root }}
-- command: ${KUBECTL} patch {{ $resource.KindGroup }}/{{ $resource.Name }} --type=merge -p '{"spec":{"forProvider":{{ $resource.UpdateParameter }}}}'
-{{- end }}
-{{- end }}
-{{- end }}
+apiVersion: chainsaw.kyverno.io/v1alpha1
+kind: Test
+metadata:
+  name: update
+spec:
+  timeouts:
+    apply: 1m
+    assert: {{ .TestCase.Timeout }}
+    exec: 1m
+  steps:
+  - name: Update Root Resource
+    description: |
+      Update the root resource by using the specified update-parameter in annotation.
+      Before updating the resources, the status conditions are cleaned.
+    try:
+    {{- range $resource := .Resources }}
+    {{- if eq $resource.KindGroup "secret." -}}
+      {{continue}}
+    {{- end -}}
+    {{- if not $resource.Namespace }}
+    {{- if $resource.Root }}
```

Comment:
Here again we leave the responsibility to the user. We have example-id annotation and it is related to update step. I think this is fair. I mean not checking this and expecting the right usage from user (because it is not failing in the other case, at worst case the update test will be skipped). But as you mentioned logging is a good idea. I added a log message when we skip the update step.